velox/common/caching/StringIdMap.cpp

```diff
@@ -43,13 +43,126 @@ void StringIdMap::release(uint64_t id) {
   }
 }
 
+#define VELOX_STRINGIDMAP_PREFIX "[StringIdMap] "
+#define VELOX_STRINGIDMAP_ERROR_LOG LOG(ERROR) << VELOX_STRINGIDMAP_PREFIX
+
+void StringIdMap::checkAndReportIntegrity(uint64_t id) const {
+  if (idToString_.size() != stringToId_.size()) {
+    VELOX_STRINGIDMAP_ERROR_LOG << fmt::format(
+        "idToString size {}. stringToId size {}. ",
+        idToString_.size(),
+        stringToId_.size());
+  }
+
+  if (idToString_.size() != lastId_) {
+    VELOX_STRINGIDMAP_ERROR_LOG << fmt::format(
+        "idToString {} element(s) were erased. ", lastId_ - idToString_.size());
+  }
+
+  if (stringToId_.size() != lastId_) {
+    VELOX_STRINGIDMAP_ERROR_LOG << fmt::format(
+        "stringToId {} elements were erased. ", lastId_ - stringToId_.size());
+  }
+
+  std::vector<std::pair<uint64_t, std::string>> missingFiles;
+  uint64_t pinnedSize{0};
+  for (auto it = stringToId_.begin(); it != stringToId_.end(); ++it) {
+    pinnedSize += it->first.length();
+    if (it->second == id) {
+      VELOX_STRINGIDMAP_ERROR_LOG << fmt::format(
+          "Found missing id {} in stringToId with file \"{}\".", id, it->first);
+    }
+    auto idEntry = idToString_.find(it->second);
```

Comment:
Why do you need this check ? I would imagine we never find it->second in idToString_ map.

Reply:
The idea is to check if the two maps are consistent. Yes, normally you wouldn't find it if everything went well. But we are looking if we find files in `stringToId_` that have an id which is not in  `idToString_`. 

The understanding was to check if the two maps are consistent. This would indicate a mismatch.